phoenix-core/src/main/java/org/apache/phoenix/util/MetaDataUtil.java

```diff
@@ -122,7 +124,36 @@
             ColumnFamilyDescriptorBuilder.KEEP_DELETED_CELLS,
             ColumnFamilyDescriptorBuilder.REPLICATION_SCOPE);
 
-    public static boolean areClientAndServerCompatible(long serverHBaseAndPhoenixVersion) {
+    public static class ClientServerCompatibility {
+
+        long serverVersion;
+        int errorCode;
+        boolean isCompatible;
+
+        ClientServerCompatibility(long serverVersion) {
+            this.serverVersion = serverVersion;
+            this.errorCode = 0;
+            this.isCompatible = true;
```

Comment:
Why is this true by default? I think it's safer for it to default to false (you don't have to assign it since primitive boolean defaults to false).